rama-http/src/layer/auth/require_authorization.rs

```diff
@@ -304,12 +353,43 @@ pub struct AuthorizeContext<C> {
 }
 
 impl<C> AuthorizeContext<C> {
-    pub(crate) fn new(credential: C) -> Self {
```

Comment:
Why do you make this `pub`? Seems you only use it in tests. Can this be reverted to `pub(crate)` again?